### PR TITLE
Firefox: Do not unobserve before observe for viewability

### PR DESCRIPTION
**Pull request by hghazzi**
### This PR will...

Keep track of observed players to avoid observing the same player multiple times, due to async calls.

### Why is this Pull Request needed?

Now that Firefox supports the Intersection Observer API, there exists a bug in v55 that breaks observing if you unobserve before ever observing

#### Addresses Issue(s):

JW8-476

### CODE REVIEW

File: src/js/view/utils/views-manager.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import activeTab from 'utils/active-tab';
 import { requestAnimationFrame, cancelAnimationFrame } from 'utils/request-animation-frame';
 
 const views = [];
+const observed = [];
```
Comment (robwalch) [inline src/js/view/utils/views-manager.js]: This should be an object, not an array.

File: src/js/view/utils/views-manager.js [hunk 2]
```diff
@@ -77,14 +78,21 @@ export default {
...
+            return;
+        }
+
+        observed[container.id] = true;
         intersectionObserver.observe(container);
     },
     unobserve(container) {
-        if (intersectionObserver) {
+        if (intersectionObserver && observed[container.id]) {
+            observed[container.id] = false;
```
Comment (robwalch) [inline src/js/view/utils/views-manager.js]: Use `delete`.

File: test/unit/views-manager-test.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        intersectionObserver = viewsManager.getIntersectionObsever();
+        intersectionObserver.observe = sinon.spy();
+        intersectionObserver.unobserve = sinon.spy();
+    });
+
+    describe('intersectionObserver', function() {
+
+        it('should observe & unobserve the same container', function() {
+            const container = document.createElement('div');
+            container.id = 1;
```
Comment (robwalch) [inline test/unit/views-manager-test.js]: Element.ids are strings not numbers

File: src/js/view/utils/views-manager.js [hunk 4]
```diff
@@ -77,14 +78,21 @@ export default {
...
     },
     unobserve(container) {
-        if (intersectionObserver) {
+        if (intersectionObserver && observed[container.id]) {
+            delete observed[container.id];
             intersectionObserver.unobserve(container);
         }
+    },
+    getIntersectionObsever() {
+        return intersectionObserver;
```
Comment (dYale) [inline src/js/view/utils/views-manager.js]: Obsever 😨

**Pull request merged**